test/bats/conjur.bats

```diff
@@ -0,0 +1,162 @@
+  helm install conjur-csi-provider cyberark/conjur-k8s-csi-provider --wait --timeout ${WAIT_TIME}s --namespace=kube-system \
+    --set providerServer.image.tag=latest
+} 
+
+@test "setup conjur" { 
+  # Create Conjur account and store admin API key
+  admin_api_key="$(kubectl exec deployment/conjur-conjur-oss \
+    --namespace=$CONJUR_NAMESPACE \
+    --container=conjur-oss \
+    -- conjurctl account create $CONJUR_ACCOUNT | grep API | awk '{print $5}')"
+
+  # Create a Conjur CLI pod
+  docker pull cyberark/conjur-cli:8
+  kind load docker-image cyberark/conjur-cli:8
+  kubectl run conjur-cli-pod --image=cyberark/conjur-cli:8 --namespace=$CONJUR_NAMESPACE --command -- sleep infinity
+  kubectl wait --for=condition=ready --timeout=${WAIT_TIME}s --namespace=$CONJUR_NAMESPACE pod/conjur-cli-pod
+
+  # Get values required by authn-jwt authenticator
+  ISSUER=$(kubectl get --raw /.well-known/openid-configuration | jq -r .issuer)
+  JWKS='{"type":"jwks","value":'$(kubectl get --raw /openid/v1/jwks)'}'
+
+  # Copy files into CLI container
+  kubectl -n "${CONJUR_NAMESPACE}" cp $BATS_TESTS_DIR/policy conjur-cli-pod:/policy -c conjur-cli-pod
+
+  # Exec into CLI container
+  kubectl -n "${CONJUR_NAMESPACE}" exec conjur-cli-pod -- /bin/sh -c "
+  set -x
+  # Initialise CLI and login
+  echo yes | conjur init -u 'https://$CONJUR_URL' -a '$CONJUR_ACCOUNT' --self-signed
+  conjur login -i admin -p $admin_api_key
+
+  # Apply policy
+  conjur policy replace -b root -f /policy/host.yaml
+  conjur policy load -b root -f /policy/authn-jwt.yaml
+  conjur policy load -b root -f /policy/variables.yaml
+  # Set test secret values
+  conjur variable set -i db-credentials/username -v '$EXPECTED_USERNAME'
+  conjur variable set -i db-credentials/password -v '$EXPECTED_PASSWORD'
+  # Set variable values on authenticator
+  conjur variable set -i conjur/authn-jwt/kube/public-keys -v '$JWKS'
+  conjur variable set -i conjur/authn-jwt/kube/issuer -v '$ISSUER'
+  "
+}
+
+@test "deploy conjur secretproviderclass crd" {
+  CONJUR_POD=$(kubectl get pods --namespace=conjur -l "app=conjur-oss" -o=jsonpath='{.items[0].metadata.name}')
+  export CONJUR_SSL_CERT=$(kubectl exec --namespace=conjur -c conjur-oss $CONJUR_POD  -- sh -c "openssl s_client -showcerts -connect $CONJUR_URL:443 </dev/null 2>/dev/null | sed -n '/-----BEGIN CERTIFICATE-----/,/-----END CERTIFICATE-----/p'")
+  envsubst < $BATS_TESTS_DIR/conjur_v1_secretproviderclass.yaml | sed '/^ *-----BEGIN CERTIFICATE-----/,$s/^/      /' | kubectl apply --namespace=kube-system -f -
+
+  kubectl wait --for condition=established --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/conjur -o yaml --namespace=kube-system | grep conjur"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+}
+
+@test "CSI inline volume test with pod portability" {
+  kubectl apply --namespace=kube-system -f $BATS_TESTS_DIR/pod-secrets-store-inline-volume-crd.yaml
```

Comment:
Does the workload need to be in `kube-system` namespace? If someone looks at this file for reference, I don't want them to assume this is a requirement. Could we use any namespace other than `kube-system` for the workloads being deployed?

Reply:
I've moved it test workloads to the `default` namespace